File: code/bit_manipulation/bit_division/BitDivision.java
import java.util.Scanner;
// Part of Cosmos by OpenGenus Foundation
class BitDivision
{
	static int division(int divident, int divisor)
	{
		int curr=1;
		int result=0;	
		if ( divisor >divident)
    	    return 0;

    	else if ( divisor ==divident)
    	    return 1;

    	while (divisor <divident) 
    	{
    	    divisor <<= 1;
    	    curr <<= 1;
    	}

    	divisor >>= 1;
    	curr >>= 1;

  	  	while (curr!=0) 
  	  	{
        	if (divident >= divisor) 
        	{
        	   divident -= divisor;
           	 	result |= curr;
        	}
        	curr >>= 1;
        	divisor >>= 1;
    	}
    	return result;
	}

	public static void main(String[]args)
	{
		Scanner scan=new Scanner(System.in);
		//a -> represents divident
		int a=scan.nextInt();
		//b -> represents divisor
		int b=scan.nextInt();
		System.out.println(division(a,b));
	}
}

File: cosmos-master/code/bit-manipulation/bit_division/bitDivision.c
#include <stdio.h>
// Part of Cosmos by OpenGenus Foundation
unsigned bitDivision(unsigned numerator, unsigned denominator) {

    unsigned current = 1;
    unsigned answer=0;

    if ( denominator > numerator)
        return 0;

    else if ( denominator == numerator)
        return 1;

    while (denominator < numerator) {
        denominator <<= 1;
        current <<= 1;
    }

    denominator >>= 1;
    current >>= 1;

    while (current!=0) {
        if ( numerator >= denominator) {
            numerator -= denominator;
            answer |= current;
        }
        current >>= 1;
        denominator >>= 1;
    }
    return answer;
}

void test(){
    int numerator = 23, denominator = 5;
    printf("%d\n", bitDivision(numerator,denominator));
}

int main() {
    test();
    return 0;
}
